lesson08/assignment/sp1/src/components/Jumbotron.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <section>
+    <h2>{{jumboprop.jumboTagline}}</h2>
+    <h3 class="lead">{{jumboprop.jumboSubTag}}</h3>
+    <!-- if jumboGraphics array item present, render on page -->
+    <div v-if="jumboprop.jumboGraphics" class="CTAcontainer">
+      <a href="https://www.twitter.com/" target="blank"><i class="fab fa-twitter fa-5x twitterCTA"></i></a>
+      <a class="button" href="https://www.twitter.com" target="_blank">{{jumboprop.jumboGraphics}}</a>
+    </div>
+    <p>{{ jumboprop.jumboCopy }}</p>
+    <!-- if jumboButton array item present, render on page -->
+    <!-- use router-link for link to other views -->
+    <div class="jumboButton-container"><router-link class="jumboButton" v-if="jumboprop.jumboButton" to="/map">{{ jumboprop.jumboButton }}</router-link></div>
+  </section>
+</template>
+
+<script>
+
+export default {
+  name: 'jumbotron',//name of file, component, html tag
+  props: {
+    //jumboprop is an object. it is attached to the array item in jumbotronArray with a v-bind.
+    jumboprop: Object,
+    //my prop name : data type
+  }
```

Comment:
It would be a better use of prop validation if jubmoSubTag, jumboTagline, jumboGraphics, jumboCopy, and jumboButton were each a prop. That way each could have their own validation rules. Receiving just an object as prop and using its attributes isn't as foolproof.

Reply:
Interesting! I will have to investigate this!